test/unit/sdam/server.test.ts

```diff
@@ -59,7 +58,11 @@ describe('Server', () => {
   describe('#handleError', () => {
     let server: Server, connection: Connection | undefined;
     beforeEach(() => {
-      server = new Server(new Topology([], {} as any), new ServerDescription('a:1'), {} as any);
+      server = new Server(
+        topologyWithPlaceholderClient([], {} as any),
```

Comment:
I'll just comment once but apply to all usages, can we make the type on the helper one where we don't need to cast to `as any`. maybe: `Partial<TopologyOptions>`?